spec/models/solidus_bolt/payment_source_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'spec_helper'
+
+RSpec.describe SolidusBolt::PaymentSource, type: :model do
+  let(:column_list) {
+    [
+      'id',
+      'payment_method_id',
+      'transaction_id',
+      'transaction_reference',
+      'transaction_type',
+      'processor',
+      'date',
+      'transaction_status',
+      'amount',
+      'currency',
+      'created_at',
+      'updated_at'
+    ]
+  }
+
+  describe 'Check Model Integrity' do
+    it 'has correct table name' do
+      expect(described_class.table_name).to eq 'solidus_bolt_payment_sources'
+    end
+
+    it 'has correct attributes' do
+      expect(described_class.new.attributes.keys).to eq column_list
+    end
+  end
+
+  describe 'transaction_reference' do
+    it 'raises an error when transaction_reference is not present' do
+      object = described_class.new
+      expect { object.save! }.to raise_error ActiveRecord::RecordInvalid
```

Comment:
what about testing with `.valid?` and checking the value of `.errors.messages`

Reply:
I don't understand this one

Can you please explain a little more how exactly do you want this to be tested ?

Reply:
well, the idea is that if in the future we add a new validation we won't know if this test is raising error `ActiveRecord::RecordInvalid` due to the original validation or a new one
so instead of writing the test as you did I'm proposing to writing it like the following:
```
       object = described_class.new
       object.valid?
       expect(object.errors.message).to eq(error_message)
```
where we could use `eq` or `include` and the `error_message` woould be the one we get from the validation

Reply:
Makes sense.
I'll modify it

Reply:
Done